tests/settings_tests/tests.py

```diff
@@ -567,3 +571,197 @@ def decorated_function():
+
+        # Empty string settings should be used if we want to ensure that
+        # SCRIPT_NAME is identical to the setting.
+        ScriptNameTestCase(
+            script_name='/somesubpath/',
+            initial_static_url='',
+            final_static_url='/somesubpath/',
+            initial_media_url='',
+            final_media_url='/somesubpath/', ),
+
+        # ``None`` values for the settings are just returned as is.
+        ScriptNameTestCase(
+            script_name='/somesubpath/',
+            initial_static_url=None,
+            final_static_url=None,
+            initial_media_url=None,
+            final_media_url=None, ),
+
+        # SCRIPT_NAME not set so no prefixation occurs
+        ScriptNameTestCase(
+            script_name=None,
+            initial_static_url='static/',
+            final_static_url='static/',
+            initial_media_url='media/',
+            final_media_url='media/', ),
+
+        # SCRIPT_NAME is empty string and no forward slashes: no prefixation
+        ScriptNameTestCase(
+            script_name='',
+            initial_static_url='static/',
+            final_static_url='static/',
+            initial_media_url='media/',
+            final_media_url='media/', ),
+
+        # SCRIPT_NAME is empty string and there are forward slashes: no prefixation
+        ScriptNameTestCase(
+            script_name='',
+            initial_static_url='/static/',
+            final_static_url='/static/',
+            initial_media_url='/media/',
+            final_media_url='/media/', ),
+
+        # SCRIPT_NAME is forward slash and no forward slashes: no prefixation
+        ScriptNameTestCase(
+            script_name='/',
+            initial_static_url='static/',
+            final_static_url='static/',
+            initial_media_url='media/',
+            final_media_url='media/', ),
+
+        # SCRIPT_NAME is forward slash and there are forward slashes: no prefixation
+        ScriptNameTestCase(
+            script_name='/',
+            initial_static_url='/static/',
+            final_static_url='/static/',
+            initial_media_url='/media/',
+            final_media_url='/media/', ),
+
+    )
+
+    def setUp(self):
+        clear_script_prefix()
+
+    def tearDown(self):
+        clear_script_prefix()
+
+    @staticmethod
+    def set_script_name(val):
+        clear_script_prefix()
+        if val is not None:
+            set_script_prefix(val)
+
+    def test_add_script_prefix(self):
+        """Perform all of the tests encoded in ``script_name_test_cases``."""
+        for case in self.script_name_test_cases:
+            s = LazySettings()
+            s.configure(
+                STATIC_URL=case.initial_static_url,
+                MEDIA_URL=case.initial_media_url, )
+            self.set_script_name(case.script_name)
+            self.assertEqual(s.STATIC_URL, case.final_static_url)
+            self.assertEqual(s.MEDIA_URL, case.final_media_url)
```

Comment:
This should use `self.subTest`, I also don't think that we need a `namedtuple` and `self.script_name_test_cases` variable, e.g.
```python
tests = (
    # SCRIPT_NAME ends with no slash, settings start with slashes.
    ('/somesubpath', '/static/', '/somesubpath/static/', '/media/', '/somesubpath/media/'),
    # SCRIPT_NAME ends with no slash, settings start with no slashes.
    ('/somesubpath', 'static/', '/somesubpath/static/', 'media/', '/somesubpath/media/'),
    ...
)
for script_name, initial_static_url, final_static_url, initial_media_url, final_media_url in tests:
    with self.subTest(...):
        ...
```

Reply:
done